muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUserSchoolDataBridge.java

```diff
@@ -1424,15 +1425,19 @@ public BridgeResponse<StudentContactLogEntryCommentRestModel> createStudentConta
     }
     BridgeResponse<StudentContactLogEntryCommentRestModel> response = pyramusClient.responsePost(String.format("/students/students/%d/contactLogEntries/%d/comments", studentId, entryId), Entity.entity(payload, MediaType.APPLICATION_JSON), StudentContactLogEntryCommentRestModel.class);
 
-    boolean hasImage = false;
-    if (response.getEntity() != null ) {
-      if (response.getEntity().getCreatorId() != null) {
-        response.getEntity().setCreatorId(toUserEntityId(response.getEntity().getCreatorId()));
-        
-        UserEntity userEntity = userEntityController.findUserEntityById(toUserEntityId(response.getEntity().getCreatorId()));
-        hasImage = userEntityFileController.hasProfilePicture(userEntity);
+    if (response.getEntity() != null && response.getEntity().getCreatorId() != null) {
+      response.getEntity().setCreatorId(toUserEntityId(response.getEntity().getCreatorId()));
+
+      boolean hasImage = false;
+      if (response.getEntity() != null ) {
+        if (response.getEntity().getCreatorId() != null) {
+          response.getEntity().setCreatorId(toUserEntityId(response.getEntity().getCreatorId()));
+          
+          UserEntity userEntity = userEntityController.findUserEntityById(toUserEntityId(response.getEntity().getCreatorId()));
+          hasImage = userEntityFileController.hasProfilePicture(userEntity);
```

Comment:
Same as with the previous method; null checks and setting creator id done twice, `userEntity` not null checked.